[PM-19305] Enforce session timeout policy

LRNcardozoWDF opened the pull request:
## 🎟️ Tracking

<!-- Paste the link to the Jira or GitHub issue or otherwise describe / point to where this change is coming from. -->
[PM-19305](https://bitwarden.atlassian.net/browse/PM-19305)

## 📔 Objective

<!-- Describe what the purpose of this PR is, for example what bug you're fixing or new feature you're adding. -->
Allow organizations to enforce session timeout policy in the iOS app.

## 📸 Screenshots

<!-- Required for any UI changes; delete if not applicable. Use fixed width images for better display. -->

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes


[PM-19305]: https://bitwarden.atlassian.net/browse/PM-19305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

matt-livefront commented on BitwardenShared/UI/Platform/Settings/Settings/AccountSecurity/AccountSecurityState.swift:
```diff
@@ -169,25 +179,37 @@ struct AccountSecurityState: Equatable {
-                policyTimeoutAction.localizedName,
-            )
-        } else {
-            Localizations.vaultTimeoutPolicyInEffect(
```
⛏️ Can these localizations be removed if we don't need them anymore?

LRNcardozoWDF commented on BitwardenShared/UI/Platform/Settings/Settings/AccountSecurity/AccountSecurityState.swift:
```diff
@@ -208,4 +230,64 @@ struct AccountSecurityState: Equatable {
+                    guard let time = option.minutesValue else { return false }
+                    return time <= value
+                } else {
+                    return true
```
I test the results of the filtering [here](https://github.com/bitwarden/ios/pull/2127/files#diff-c5216fde15d073221e7efa428af03b5c41b9bb823bca63a97a97d28eacfad2e5R116-R125) . I can also try and return the array with the right options given a type:
```suggestion
var availableTimeoutOptions: [SessionTimeoutValue] {
        switch policyTimeoutType {
        case .never:
            SessionTimeoutValue.allCases
        case .onAppRestart:
            [
                .immediately,
                .oneMinute,
                .fiveMinutes,
                .fifteenMinutes,
                .thirtyMinutes,
                .oneHour,
                .fourHours,
                .onAppRestart,
                .custom(-100),
            ]
        case .immediately:
            [.immediately]
        case .custom:
            SessionTimeoutValue.allCases.filter { option in
                option.isCustomPlaceholder || (option.minutesValue.map { $0 <= policyTimeoutValue } ?? false)
            }
        case nil:
            if policyTimeoutValue > 0 {
                SessionTimeoutValue.allCases.filter { option in
                    option.isCustomPlaceholder || (option.minutesValue.map { $0 <= policyTimeoutValue } ?? false)
                }
            } else {
                SessionTimeoutValue.allCases
            }
        }
    }
```

LRNcardozoWDF commented on BitwardenShared/Core/Vault/Services/SyncServiceTests.swift:
```diff
@@ -157,22 +163,54 @@ class SyncServiceTests: BitwardenTestCase {
+
+        try await subject.fetchSync(forceSync: false)
+
+        XCTAssertEqual(stateService.timeoutAction["1"], .logout)
```
Claude, I added that validation in the line below.

matt-livefront commented on BitwardenKit/UI/Platform/Application/Views/SettingsPickerField+ViewInspectorTests.swift:
```diff
@@ -0,0 +1,159 @@
+
+        // Update the bound value
+        pickerValue = testValue
+        XCTAssertEqual(pickerValue, testValue)
```
🤔 This doesn't seem like it's actually testing the binding, since it's setting pickerValue to testValue and then ensuring they are equal. Maybe we could see if there's a way to access the date picker and update the internal picker value and ensure the binding updates?

LRNcardozoWDF commented:
**Overall Assessment:** APPROVE

Reviewed commit 882294e4 "pm-19305 Fix pr comments" which includes UI component updates and localization changes for vault timeout policy messaging.

## Summary

This commit makes targeted improvements to UI components and updates localization strings to clarify the distinction between "default" and "maximum" session timeout messaging:

**Changes reviewed:**
- SettingsPickerField: Added support for empty title display with adjusted styling
- BitwardenMenuField & BitwardenToggle: Improved footer padding consistency  
- AccountSecurityState: Refactored availableTimeoutOptions to use computed property
- Localization: Added "maximum session timeout" strings while preserving "default" for type-based timeouts
- Tests: Updated to reflect new localization strings
- Snapshots: Re-recorded to reflect UI changes

## Detailed Findings

### Positive Observations

**1. Semantic clarity in localization (AccountSecurityState.swift:157-180)**
Good distinction between:
- `yourOrganizationHasSetTheMaximumSessionTimeoutToX/XAndY` - Used for custom duration-based timeouts (1 hour, 2h 30m, etc.)
- `yourOrganizationHasSetTheDefaultSessionTimeoutToX` - Preserved for type-based timeouts (.never, .onAppRestart)

This correctly reflects that "maximum" applies to measurable durations while "default" applies to timeout behaviors.

**2. Code quality improvement (AccountSecurityState.swift:124-146)**
The refactoring of `availableTimeoutOptions` from a private helper method to a direct computed property is cleaner and more idiomatic Swift. The switch-based implementation is easier to read and maintain.

**3. UI consistency (BitwardenMenuField.swift, BitwardenToggle.swift)**
Footer padding adjustments create consistent visual hierarchy:
- Divider consistently padded from leading edge only
- Footer content properly inset with horizontal padding
- Maintains design system alignment across components

**4. SettingsPickerField flexibility (SettingsPickerField.swift:39-67)**
Added support for empty title scenario with appropriate layout adjustments:
- Conditional title display with `if !title.isEmpty`
- Dynamic text alignment based on title presence
- Conditional vertical padding
- Value badge styling (background, corner radius)

### Areas of Attention

**1. Disabled snapshot tests (SettingsPickerField+SnapshotTests.swift)**
All snapshot tests in this file are prefixed with `disable`, preventing them from running. This appears to be intentional (likely due to `.test-simulator-device-name` configuration requirements), but worth confirming:
- Lines 27, 39, 51, 63, 76, 89, 102: All tests prefixed with `disabletest_`

If these tests are meant to run in CI with the correct simulator, the prefix should be removed.

**2. Whitespace-only change (PolicyServiceTests.swift:44)**
Line 44 has a trailing whitespace removal - this is good hygiene but could have been in a separate formatting commit. Not blocking.

**3. Preview additions (SettingsPickerField.swift:133-139, BitwardenMenuField.swift:431-440, BitwardenToggle.swift:178-183)**
Good additions for developer experience, but consider whether these should have DocC documentation explaining when to use empty-title variant.

## Test Coverage

- ✅ Unit tests updated (AccountSecurityProcessorTests.swift) 
- ✅ Snapshot images re-recorded for SettingsPickerField
- ✅ Test localization strings updated
- ⚠️ Snapshot tests currently disabled - verify if intentional

## Code Style Compliance

- ✅ Follows Swift code style guidelines
- ✅ Proper use of guard clauses and optionals
- ✅ DocC documentation preserved where appropriate
- ✅ Consistent indentation and formatting

## Security Considerations

No security implications identified. Changes are UI-focused with appropriate state management through existing patterns.

## Recommendation

**APPROVE** - This is a well-structured commit addressing PR feedback with appropriate semantic distinctions in user-facing messaging. The code quality improvements and UI consistency changes are positive. The only minor concern is the disabled snapshot tests, which should be verified as intentional.
